### ISSUE TITLE
Add more complex examples to tutorials showcasing differentiability

### ISSUE THREAD

**Issue by gomezzz** [esa/torchquad#125]
# Feature
## Desired Behavior / Functionality
torchquad allows fully differentiable numerical integration. This can enable neural network training through integrals. This capability deserves a dedicated example. There is a example on the gradient computations in the [docs](https://torchquad.readthedocs.io/en/main/tutorial.html#computing-gradients-with-respect-to-the-integration-domain) already. However, training a simple neural network on e.g. a gravitational potential or similar problem could be a cool example?

## What Needs to Be Done
- [ ] Find a nice example to showcase neural network training / optimization capabilities due to torchquad's differentiability
- [ ] Add to docs
